text/0474-event-bridge-scheduler-l2.md

```diff
@@ -0,0 +1,569 @@
+3. **Schedule Group**: A schedule group is an Amazon EventBridge Scheduler resource that you use to organize your schedules. Your AWS account comes with a default scheduler group. A new schedule will always be added to a scheduling group. If you do not provide a scheduling group to add to, it will be added to the default scheduling group. You can create up to 500 schedule groups in your AWS account. Groups can be used to organize the schedules logically, access the schedule metrics and manage permissions at group granularity (see details below). Scheduling groups support tagging: with EventBridge Scheduler, you apply tags to schedule groups, not to individual schedules to organize your resources.
+
+This module is part of the [AWS Cloud Development Kit](https://github.com/aws/aws-cdk) project. It allows you to define Event Bridge Schedules.
+
+
+## Defining a schedule 
+
+```ts
+const target = new targets.LambdaInvoke(props.func, {
+    input: ScheduleTargetInput.fromObject({
+        "payload": "useful"
+    })
+});
+    
+const schedule = new Schedule(this, 'Schedule', {
+    schedule: ScheduleExpression.rate(Duration.minutes(10)),
+    target,
+    description: 'This is a test schedule that invokes lambda function every 10 minutes.',
+});
+```
+
+### Schedule Expressions
+
+You can choose from three schedule types when configuring your schedule: rate-based, cron-based, and one-time schedules. 
+
+Both rate-based and cron-based schedules are recurring schedules. You configure each recurring schedule type using a schedule expression for the type of schedule you want to configure, and specifying a time zone in which EventBridge Scheduler evaluates the expression with `scheduleTimeZone` parameter.
+
+```ts
+const rateBasedSchedule = new Schedule(this, 'Schedule', {
+    schedule: ScheduleExpression.rate(Duration.minutes(10)),
+    target,
+    description: 'This is a test rate-based schedule',
+});
+
+const cronBasedSchedule = new Schedule(this, 'Schedule', {
+    schedule: ScheduleExpression.cron({ minute: '0', hour: '23', day: '20', month: '11' }),
+    target,
+    scheduleTimeZone: TimeZone.AMERICA_NEW_YORK,
+    description: 'This is a test cron-based schedule that will run at 11:00 PM, on day 20 of the month, only in November in New York timezone',
+});
+```
+
+A one-time schedule is a schedule that invokes a target only once. You configure a one-time schedule when by specifying the time, date, and time zone in which EventBridge Scheduler evaluates the schedule.
+
+```ts
+const oneTimeSchedule = new Schedule(this, 'Schedule', {
+    schedule: ScheduleExpression.at(new Date(2022, 10, 20, 19, 20, 23)),
```

Comment:
I guess this `Date` class will be defined somewhere in CDK, as we cannot use Node's built-in Date class here due to jsii limitation.

Reply:
I think I only implemented it this way when I saw another instance where Date was used in the CDK libraries. Will have a look :-)

Reply:
There are many examples of Date being used. Here is an example:

https://github.com/aws/aws-cdk/blob/main/packages/aws-cdk-lib/aws-applicationautoscaling/lib/scalable-target.ts

Reply:
TIL about that! Reading [this doc](https://aws.github.io/jsii/specification/2-type-system/), it seems Date is also  supported (a bit misleading though). I'll archive this 👍